### PR TITLE
Ht4 testing

### PR DESCRIPTION
**Pull request by alexeymatyas**
Hi @enyalux. Could you please review my home task4 implementation

### CODE REVIEW

File: src/test/java/com/epam/service/MockPoliceForceServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            public String call() throws Exception {
+                return underTest.getContactNo("");
+            }
+        });
+        executor.submit(getNumberTask);
+        executor.shutdownNow();
+
+        assertThat("Should return empty upon interruption",
+                getNumberTask.get(), equalTo(""));
+    }
```
Comment (enyalux) [inline src/test/java/com/epam/service/MockPoliceForceServiceTest.java]: Minor: you can use constant for "". You can create your own or you can use https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html

File: src/test/java/com/epam/service/MockPoliceForceServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.concurrent.*;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Created by Alexey on 03.05.2016.
+ */
+public class MockPoliceForceServiceTest {
```
Comment (enyalux) [inline src/test/java/com/epam/service/MockPoliceForceServiceTest.java]: i suggest you use Mockito to check positive(thenReturn) and negative (thenThrow) behaviour for your service.
Also you can validate if correct methods were called.

File: src/test/java/com/epam/service/MockPoliceForceServiceTest.java (hunk 1 above)
Comment (alexeymatyas) [inline src/test/java/com/epam/service/MockPoliceForceServiceTest.java]: Done

File: src/test/java/com/epam/service/MockPoliceForceServiceTest.java (hunk 2 above)
Comment (alexeymatyas) [inline src/test/java/com/epam/service/MockPoliceForceServiceTest.java]: Well, I made a confusion here naming the class with Mock... prefix. It's actually dummy service emulating network communication. So there's no reason to mock the service here as I'm testing real dummy service. Renamed the service now.

From other side this is not unit test at all but integration one:)

**Comment by enyalux** [comment]
Homework accepted - closing PR. Good job!

**Pull request closed without merging**